syntax = "proto3";

package utils;


message Relation {
    enum RelationType {
        CHILD = 0;  // Child resource
        ABOUT = 1;  // related with label (GENERATED)
        ENTITY = 2;  // related with an entity (GENERATED)
        COLAB = 3;  // related with user (GENERATED)
        OTHER = 5;  // related with something
    }
    
    RelationType relation = 5;
    RelationNode source = 6;
    RelationNode to = 7;
    string relation_label = 8;
}

message RelationNode {
    enum NodeType {
        ENTITY=0;
        LABEL=1;
        RESOURCE=2;
        USER=3;
    }
    string value = 4;
    NodeType ntype = 5;
    string subtype = 6;
}

message ExtractedText {
    string text = 1;
    map<string, string> split_text = 2;
    repeated string deleted_splits = 3;
}


message Vector {
    int32 start = 1;
    int32 end = 2;
    int32 start_paragraph = 3;
    int32 end_paragraph = 4;
    repeated float vector = 5;
}

message Vectors {
    repeated Vector vectors = 1;
}

message VectorObject {
    Vectors vectors = 1;
    map<string, Vectors> split_vectors = 2;
    repeated string deleted_splits = 3;
}
